### PR TITLE
publish test catalog to github now requires a galasa.token for authentication

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>

- Insist on the galasa.token information to push a test catalog to the galasa ecosystem
- Improve logging and errors in general
- Turn the galasa.token value into a JWT using the auth server
- Pass the JWT token on the request to update the test catalog
- Documentation for the maven plugin

This PR is in response to this issue: https://github.com/galasa-dev/projectmanagement/issues/1748

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/auth/AuthenticationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    }
+
+
+    // This test was helpful to show that the AuthenticationService code should work against a real server.
+    // To use it, you need to plug-in your own token.
+    // @Test
+    // public void testCanTargetARealEcosystem() throws Exception {
+    //     String refreshToken = "xxx";
+    //     String clientId = "xxx";
+    //     String apiServerUrlString = "https://galasa-ecosystem1.galasa.dev/api";
```
Comment (KirbyKatcher) [inline galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/auth/AuthenticationServiceTest.java:231]: I think apiServerUrlString should point to a dummy url here i.e. `https://my.server.url/api`

**Review by KirbyKatcher — APPROVED** [review]
approved based on commented change

**Comment by galasa-team** [comment]
Build successful

File: galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/auth/AuthenticationServiceTest.java (hunk 1 above)
Comment (techcobweb) [reply galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/auth/AuthenticationServiceTest.java]: Done.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by KirbyKatcher — APPROVED** [review]

**Pull request merged**